src/loader.js

```diff
@@ -5,155 +5,198 @@ let Promise = require('promise');
 let _ = require('lodash');
 let TopoSort = require('topo-sort');
 
-// We use this closure to create a .create() method which
-// returns a graphViz file for a given component directory
-function generateGraph(dir, vComp) {
-  let tsort = new TopoSort();
-
-  for (let x of _.keys(dir)) {
-    tsort.add(x, dir[x].requires || []);
+/** Check if a value is a flat value or a component definition */
+let isComponent = (def) => {
+  // Check that it's an object
+  if (typeof(def) !== 'object' && def !== null && def !== undefined) {
+    return false;
   }
-
-  for (let x of _.keys(vComp)) {
-    tsort.add(x, vComp[x].requires || []);
+  // Check that is object has a setup function
+  if (!(def.setup instanceof Function)) {
+    return false;
+  }
```

Comment:
If we want to support flat values then a non-function setup should be treated at flat value..

But I would be okay, either way... even not supporting flat values (because they are really just constants). In which case, `isComponent` should just throw if it's not a component.

@jhford, should we support flat values... I think that just doing a top-level `const MYCONSTANT = "flat value"` in the components.js file would solve the problem for constants..